crates/flux-infer/src/lean_encoding.rs

```diff
@@ -16,28 +17,30 @@ use crate::{
     lean_format::{self, LeanConstDecl, LeanSortDecl, LeanSortVar, LeanVar},
 };
 
-pub struct LeanEncoder<'genv, 'tcx, 'a> {
+pub struct LeanEncoder<'genv, 'tcx> {
     genv: GlobalEnv<'genv, 'tcx>,
-    lean_path: &'a Path,
+    lean_path: PathBuf,
     project_name: String,
     defs_file_name: String,
 }
 
-impl<'genv, 'tcx, 'a> LeanEncoder<'genv, 'tcx, 'a> {
-    pub fn new(
-        genv: GlobalEnv<'genv, 'tcx>,
-        lean_path: &'a Path,
-        project_name: String,
-        defs_file_name: String,
-    ) -> Self {
+impl<'genv, 'tcx> LeanEncoder<'genv, 'tcx> {
+    pub fn new(genv: GlobalEnv<'genv, 'tcx>, project_name: String, defs_file_name: String) -> Self {
+        let lean_path = genv
```

Comment:
I believe `cargo-flux` is in a better position to decide where to put lean files, e.g., in a multi-crate workspace, `cargo-flux` could decide to put lean files inside each crate. So, I'd add a flag to Flux to set the path and let `cargo-flux` choose what the path should be.

Reply:
Yes, that's what I'm suggesting.

In flux land, I'd accept both relative and absolute paths. If absolute, we don't do anything special. If relative, we consider it relative to the `working_dir`, so we can just append it to `tcx.sess.opts.working_dir`

In `cargo-flux`, we can pass either an absolute or a relative path, depending on which one is easier.

Reply:
Ok I think that's all done -- I *believe* that `path.join(rest)` takes care of the relative/absolute business?